src/rustdoc-json-types/lib.rs

```diff
@@ -385,21 +388,24 @@ pub enum AssocItemConstraintKind {
     Constraint(Vec<GenericBound>),
 }
 
-/// An opaque identifier for an item.
+/// An opaque identifier for an item. The integer within indexes into
+/// [`Crate::index`] to resolve it to an [`Item`], or into [`Crate::paths`]
+/// to resolve it to an [`ItemSummary`].
 ///
-/// It can be used to lookup in [`Crate::index`] or [`Crate::paths`] to resolve it
-/// to an [`Item`].
-///
-/// Id's are only valid within a single JSON blob. They cannot be used to
-/// resolve references between the JSON output's for different crates.
-///
-/// Rustdoc makes no guarantees about the inner value of Id's. Applications
-/// should treat them as opaque keys to lookup items, and avoid attempting
-/// to parse them, or otherwise depend on any implementation details.
```

Comment:
Is it worth keeping this portion of the comment?

On one hand, this comment used to be a lot more useful when the ID was of the form `2:1234` (more or less) where `2` was the crate ID and the rest could be (ab)used as a `DefId` inside that crate with better than 50-50 odds of working out. On the other hand, the ID values may change representation again in the future, and this is a solid future-proofing warning.

Reply:
Thanks for explaining how it used to be different, now I understand how the comment came to be. It used to make sense, but no more. The id is just an index into another part of the JSON. That's not an "opaque key" in any sense. And the idea of parsing an integer is silly. 

As for future-proofing: sure, this could change, but so could literally anything else in the representation. So it doesn't need explicit mention.